addon/components/popover/panel/index.ts

```diff
@@ -0,0 +1,30 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+
+import { normalizeKey } from '@showbie/backpack-ember/utils/normalize-keycode';
+
+interface PanelArgs {
+  isOpen: boolean;
+  panelElement: HTMLElement;
+  closePopover: () => void;
+}
+
+export default class Panel extends Component<PanelArgs> {
+  @action
+  handleKeyUp(event: KeyboardEvent) {
+    const { isOpen, panelElement, closePopover } = this.args;
+    const eventKey = normalizeKey(event);
+
+    switch (eventKey) {
+      case 'Escape':
+        if (!isOpen) return;
+        if (!panelElement) return;
+        if (!panelElement.contains(document.activeElement)) return;
+
+        event.preventDefault();
+        event.stopPropagation();
+        closePopover();
```

Comment:
This internal action calls the `closePopover` argument function, which in turn updates the value of `isOpen`.

Reply:
Not sure if this affects anything that you are working with right now but shouldn't this be `this.args.closePopover()`?

Reply:
Hmm, it's destructured off of `this.args` a few lines up, but I'll see if it makes a difference...